rust/scx_utils/src/topology.rs

```diff
@@ -217,10 +220,48 @@ impl Cache {
     }
 }
 
+#[derive(Debug, Clone)]
+pub struct GPU {
+    id: usize,
+    node_id: usize,
+    max_graphics_clock: usize,
+    // AMD uses CU for this value
+    max_sm_clock: usize,
+    memory: u64,
```

Comment:
Can we just `pub` these fields instead of adding accessors? That way, it's a lot easier to e.g. unpack the fields in `match` arms.